audiorecorder/src/main/java/org/odk/collect/audiorecorder/recording/AudioRecorderViewModel.kt

```diff
@@ -1,46 +1,14 @@
 package org.odk.collect.audiorecorder.recording
 
-import android.app.Application
-import android.content.Intent
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.ViewModel
-import androidx.lifecycle.ViewModelProvider
-import org.odk.collect.audiorecorder.recording.AudioRecorderService.Companion.ACTION_CANCEL
-import org.odk.collect.audiorecorder.recording.AudioRecorderService.Companion.ACTION_START
-import org.odk.collect.audiorecorder.recording.AudioRecorderService.Companion.ACTION_STOP
 import java.io.File
-import javax.inject.Inject
 
-internal class AudioRecorderViewModel(private val application: Application, private val recordingRepository: RecordingRepository) : ViewModel() {
-
-    val recording: LiveData<File?> = recordingRepository.getRecording()
-
-    fun start() {
-        application.startService(
-            Intent(application, AudioRecorderService::class.java).apply { action = ACTION_START }
-        )
-    }
-
-    fun stop() {
-        application.startService(
-            Intent(application, AudioRecorderService::class.java).apply { action = ACTION_STOP }
-        )
-    }
-
-    fun cancel() {
-        application.startService(
-            Intent(application, AudioRecorderService::class.java).apply { action = ACTION_CANCEL }
-        )
-    }
-
-    fun endSession() {
-        recordingRepository.clear()
-    }
-
-    class Factory @Inject constructor(private val application: Application, private val recordingRepository: RecordingRepository) : ViewModelProvider.Factory {
-
-        override fun <T : ViewModel?> create(modelClass: Class<T>): T {
-            return AudioRecorderViewModel(application, recordingRepository) as T
-        }
-    }
+abstract class AudioRecorderViewModel : ViewModel() {
```

Comment:
The thinking here is that `collect_app` doesn't need to know how to construct `AudioRecorderViewModel`